fe/fe-core/src/main/java/org/apache/doris/analysis/SlotRef.java

```diff
@@ -134,8 +134,16 @@ public boolean columnEqual(Expr srcExpr) {
             srcTableName = srcSlotRef.getTableName();
         }
         TableName thisTableName = tblName;
-        if (thisTableName == null && desc != null) {
-            thisTableName = getTableName();
+        if (desc != null && desc.getParent() != null && desc.getParent().getRef() != null) {
+            thisTableName = desc.getParent().getRef().getName();
+        } else if (!desc.getSourceExprs().isEmpty()) {
+            List<Expr> sourceExprs = desc.getSourceExprs();
+            Expr sourceExpr = sourceExprs.get(0);
+            if (((SlotRef) sourceExpr).desc != null && ((SlotRef) sourceExpr).desc.getParent() != null) {
```

Comment:
This value does not have to be slotREF

Reply:
If sourceExpr didn't instanceof SlotRef, I can't get db.name(default_cluster:db_name.table_name), is it right?

Reply:
But source expr may not be slotref, it may be an expression of a+b